core/src/main/java/io/questdb/network/AbstractIODispatcher.java

```diff
@@ -212,13 +225,22 @@ public void setup() {
     private void addPending(int fd, long timestamp) {
         // append pending connection
         // all rows below watermark will be registered with epoll (or similar)
+        final Socket socket = socketFactory.newInstance(nf, fd, LOG);
```

Comment:
the connect/disconnect flow was GC-free before Socket is intoroduced. The contexts are pooled by factory, while sockets are not. It might be that context needs to have a reusable socket as context itself is reusable. Rather than creating new socket instance on each connection

Reply:
I considered pooling the sockets, but it would only make a difference (if any) when someone fires queries over lots of one-off connections. But as we want to keep it litter-free, let me introduce a reusable socket that would be managed by the context.

Reply:
we already have (a lot) of use cases where ILP clients connect-send-disconnect for each metric. This does go over "heavy" path on ILP but we should stop making this path heavier.

In other words, connect-do(somthing)-disconnect is a widespread pattern

Reply:
Done in 5ee92af47fde755e6e35cfd805020f8dc848e34f